Signal/src/ViewControllers/OWSTableViewController.h

```diff
@@ -0,0 +1,56 @@
+//
+//  Copyright (c) 2017 Open Whisper Systems. All rights reserved.
+//
+
+#import <UIKit/UIKit.h>
+
+@class OWSTableItem;
+@class OWSTableSection;
+
+@interface OWSTableContents : NSObject
+
+@property (nonatomic) NSString *title;
+
+- (void)addSection:(OWSTableSection *)section;
+
+@end
+
+#pragma mark -
+
+@interface OWSTableSection : NSObject
+
+@property (nonatomic) NSString *title;
+
++ (OWSTableSection *)sectionWithTitle:(NSString *)title
+                                items:(NSArray *)items;
+
+- (void)addItem:(OWSTableItem *)item;
+
+@end
+
+#pragma mark -
+
+typedef NS_ENUM(NSInteger, OWSTableItemType) {
```

Comment:
Well, I've been debating that.  I'm certainly going to rework this, but I've been waiting until I get a chance to do a solid second pass on these classes.  Eventually, we'll definitely want some kinds of special items (such as subgroup rows that when selected present a nested set of items in a separate view).  If you'd like, we can discuss.